test/schemas/test_parametrization.py

```diff
@@ -133,3 +133,44 @@ def test_(request, case):
     result = testdir.runpytest("-v", "-s")
     result.assert_outcomes(passed=1)
     result.stdout.re_match_lines([r"Hypothesis calls: 1"])
+
+
+def test_recursive_dereference(testdir):
+    # When a given parameter contains a JSON reference
+    testdir.make_test(
+        """
+@schema.parametrize(max_examples=1)
+def test_(request, case):
+    request.config.HYPOTHESIS_CASES += 1
+    assert case.path == "/v1/users"
+    assert case.method == "GET"
+    assert_int(case.query["object"]["id"])
+""",
+        **as_param({"schema": {"$ref": "#/definitions/ObjectRef"}, "in": "query", "name": "object"}),
+        definitions={
+            "ObjectRef": {"type": "object", "properties": {"id": {"$ref": "#/definitions/SimpleIntRef"}}},
+            "SimpleIntRef": integer(name="id"),
+        },
+    )
+    # Then it should be correctly resolved and used in the generated case
+    result = testdir.runpytest("-v", "-s")
+    result.assert_outcomes(passed=1)
+    result.stdout.re_match_lines([r"Hypothesis calls: 1"])
+
+
+def test_custom_func(testdir):
+    # When a given parameter contains a JSON reference
+    testdir.make_test(
+        """
+@schema.parametrize(max_examples=1)
+def test_(request, case):
+    request.config.HYPOTHESIS_CASES += 1
+    assert case.path == "/v1/users"
+    assert case.method == "GET"
+""",
+        **as_param({"parameter": {"type": "string", "format": "custom_format", "in": "query"}}),
+    )
+    # Then it should be correctly resolved and used in the generated case
```

Comment:
There could be two comments for different assertions
"Then the relevant test case should be skipped"

"And a proper message is written to the output"